pydeseq2/dds.py

```diff
@@ -60,20 +64,35 @@ class DeseqDataSet(ad.AnnData):
 
     design_factors : str or list
         Name of the columns of metadata to be used as design variables.
-        (default: ``'condition'``).
+        Interaction terms can also be used such as col1:...:colN.
+        Finally, if you wish to use a formula, you can write:
+        ``"~ col1 + ... + colN + colk1:...:colN1 + ... + colkni:...:colNni"``.
+        In this case, the formula will be parsed and the factors (interacting
+        and non-interacting) will be extracted automatically from adata or counts
+        and metadata relying on the wonderful formulaic package. Note that for
+        the moment pydeseq2 doesn't support the entirety of formulaic grammar.
+        Formulas using syntax such as 'C(values, contr.poly, levels=[10, 20, 30])'
+        or '*' are NOT supported yet as input. For precise control over categorical
+        encoding users should use ref_level argument. Every column will be interpreted
+        as categorical by design except if listed in continuous_factors.
+        (default: ``"condition"``).
 
     continuous_factors : list or None
         An optional list of continuous (as opposed to categorical) factors. Any factor
         not in ``continuous_factors`` will be considered categorical (default: ``None``).
 
-    ref_level : list or None
-        An optional list of two strings of the form ``["factor", "test_level"]``
-        specifying the factor of interest and the reference (control) level against which
-        we're testing, e.g. ``["condition", "A"]``. (default: ``None``).
+    ref_level : list[tuple[str, str]] or None
+        An optional list of tuples each with two strings of the form
+        ``("factor", "test_level")`` specifying the factor of interest and the
+        reference (control) level against which we're testing, e.g.
+        ``[("condition", "A")]``. (default: ``None``).
```

Comment:
Note that this can also be solved by formulaic, e.g. 
```python
"~ C(condition, contr.treatment(base='A'))"
```

or it can be set using pandas categoricals before. I'm not entirely sure it's worth bothering with this here.

Reply:
Thanks for the comment. I don't want to force user to master formulaic syntax if they don't plus I would like retro-compatibility